hack/e2e.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash -e
+#
+# This script is intended to be run from the CI pipeline as follows:
+#
+#   # Define secrets in the envionment
+#   source hack/e2e.sh
+#   aro_ci_setup
+#   make e2e
+
+get_random_location() {
+    array[0]="australiasoutheast"
+    array[1]="eastus"
+    array[2]="westeurope"
+
+    size=${#array[@]}
+    index=$(($RANDOM % $size))
+    echo ${array[$index]}
+}
+
+aro_ci_setup() {
+    echo "====== starting dev RP =========================="
+    go build ./cmd/aro
+    ./aro rp &
+    trap 'return_id=$?; aro_ci_teardown_handler; exit $return_id' EXIT
+    while [[ "$(curl -s -o /dev/null -w '%{http_code}' localhost:8443)" == "000" ]]; do
+        sleep 2
+    done
+
+    if $CLUSTER_CREATE; then
+        echo "====== setup subnets =========================="
+        for subnet in "$CLUSTER-master" "$CLUSTER-worker"; do
+            az network vnet subnet create \
+                -g "$RESOURCEGROUP" \
+                --vnet-name dev-vnet \
+                -n "$subnet" \
+                --address-prefixes 10.$((RANDOM & 127)).$((RANDOM & 255)).0/24 \
+                --service-endpoints Microsoft.ContainerRegistry \
+                >/dev/null
+        done
+        az network vnet subnet update \
+            -g "$RESOURCEGROUP" \
+            --vnet-name dev-vnet \
+            -n "$CLUSTER-master" \
+            --disable-private-link-service-network-policies true \
+            >/dev/null
+
+        echo "====== aro create =========================="
+        az aro create -g "$RESOURCEGROUP" --cluster-resource-group "$CLUSTER" -n "$CLUSTER" --vnet dev-vnet --master-subnet "$CLUSTER-master" --worker-subnet "$CLUSTER-worker"
```

Comment:
I had some trouble using --cluster-resource-group, will confirm if it works (I wanted to distinguish the e2e clusters from dev ones)

Reply:
what trouble? yes, please check.